[mumath] Update typing for mumath

adamzerella opened the pull request:
As previously mentioned with @amcasey this PR adds the extra type exports for `mumath`.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/33142#issuecomment-466253406
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

amcasey commented:
@adamzerella Thanks for following up!  One question: should index.d.ts be re-exporting the symbols, rather than re-declaring them?

adamzerella commented:
> @adamzerella Thanks for following up! One question: should index.d.ts be re-exporting the symbols, rather than re-declaring them?

I'm not sure what you mean? It would be better to just re-declare the `index.d.ts` definitions rather than export them? Or are you saying import the definitions from the already declared files?

amcasey commented:
I meant that you probably don't want to have the full signature of each function appearing in more than one module - that's a recipe for having them get out of sync.  The underlying library re-exports them from index.js: https://github.com/dy/mumath/blob/master/index.js

I don't normally recommend using ES6 module syntax for node modules but, in this particular case, an ES6 re-export might be appropriate: `export { x } from "module"`.  Personally, I'd follow the underlying library and put the actual signatures in the non-index files but you could also do it the other way around if you wanted to keep them all in one place.

adamzerella commented:
Completely agree, I've updated the branch - ready for another pass. To be even more true to the underlying library I've removed the default exports in favour of exporting the functions directly.

Thanks for bearing with me @amcasey 💯

amcasey commented on types/mumath/index.d.ts:
```diff
@@ -2,62 +2,17 @@
+import scale = require('./scale');
+import within = require('./within');
+
+export { clamp, closest, isMultiple, len, lerp, mod, order, precision, round, scale, within };
```
It should be equivalent but, just for consistency, I'd probably use `export = { clamp, … };`.
